src/main/interface/msp.c

```diff
@@ -1224,8 +1224,8 @@ static bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
         sbufWriteU8(dst, 0); // reserved
         sbufWriteU8(dst, currentPidProfile->vbatPidCompensation);
         sbufWriteU8(dst, currentPidProfile->setpointRelaxRatio);
-        sbufWriteU8(dst, currentPidProfile->dtermSetpointWeight);
-        sbufWriteU8(dst, 0); // reserved
+        sbufWriteU16(dst, currentPidProfile->dtermSetpointWeight);
```

Comment:
If we do it this way, all existing applications will break, which is what you were worried about.

You should rather append the value at the end. Then do a `sbufWriteU8(dst, MIN(currentPidProfile->dtermSetpointWeight, 255))` here, and legacy apps will see the real value, or 255 if it is over. This way legacy apps will still work correctly, and new apps can read the full value from the end.

Reply:
"all existing applications will break" Why? Next byte was reserved, so I replaced U8 with U16 and removed next byte. Is should be OK for applications, all following MSP bytes stays untouched this way. I can add this value also at the end, but then there will be duplicated value, one 8bit and second 16bit. Isn't cleaner solution to use reserved byte?

Reply:
See @ledvinap's comment. If you do this, and a user sets `dtermSetpointWeight` to 256 in CLI, his legacy app will show '0'. If the legacy app happens to send 0xff for the reserved byte, the resulting setpoint weight will be very high.

Reply:
Ok, now I understand. Didn't expected, that there will be anything else than 0 :)